ledger/participant-integration-api/src/main/scala/platform/store/cache/StateCache.scala

```diff
@@ -107,13 +107,12 @@ private[platform] case class StateCache[K, V](cache: Cache[K, V], registerUpdate
               removeFromPending(key)
             }
             .getOrElse(logger.error(s"Pending updates tracker for $key not registered "))
-        },
-      (err: Throwable) => {
+        }
+      }
+      .recover { case err =>
```

Comment:
Please note that now we recover all kinds of problems not just the new error type `ContractReadThroughNotFound`. I don't know if this is a problem though

Reply:
Indeed. First of all, the problems that might arise here that were previously "un-recovered" are internal to the cache. For these failures, it is actually better to recover and log since the failure of `putAsync` is not critical to the correctness guarantee of the mutable contract state cache. On the other hand, not recovering here would will stop command interpretation and report the failure to the client.